### PR TITLE
generalized Wiener

### PR DESCRIPTION
**Pull request by t4c1**
## Summary

Generalized Wiener distribution to accept Eigen expressions.

## Tests

Tested using expression testing framework.

## Side Effects
None.

## Release notes

Generalized Wiener distribution to accept Eigen expressions.

## Checklist

- [ ] Math issue #1470 

- [ ] Copyright holder: Tadej Ciglarič

    The copyright holder is typically you or your assignee, such as a university or company. By submitting this pull request, the copyright holder is agreeing to the license the submitted work under the following licenses:
      - Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
      - Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

- [ ] the basic tests are passing

    - unit tests pass (to run, use: `./runTests.py test/unit`)
    - header checks pass, (`make test-headers`)
    - dependencies checks pass, (`make test-math-dependencies`)
    - docs build, (`make doxygen`)
    - code passes the built in [C++ standards](https://github.com/stan-dev/stan/wiki/Code-Quality) checks (`make cpplint`)

- [ ] the code is written in idiomatic C++ and changes are documented in the doxygen

- [ ] the new changes are tested

### CODE REVIEW

File: test/expressions/generateExpressionTests.py [hunk 1]
```diff
@@ -176,12 +176,13 @@ def parse_signature(signature):
     "lkj_corr_lpdf": [1, None],
     "lkj_corr_log": [1, None],
     "log_diff_exp": [3, None],
-    "log_inv_logit_diff": [1.2, 0.4],
+    "log_inv_logit_diff": [1.2, 0.4],
```
Comment (bbbales2) [inline test/expressions/generateExpressionTests.py:179]: I feel like I've seen this space get removed before lol

File: test/expressions/generateExpressionTests.py [hunk 2]
```diff
@@ -237,12 +238,12 @@ def save_tests_in_files(N_files, tests):
 
 def handle_function_list(functions_input, signatures):
     """
-    Handles list of functions, splitting elements between functions and signatures.
+    Handles list of functions, splitting elements between functions and signatures.
```
Comment (bbbales2) [inline test/expressions/generateExpressionTests.py:241]: If in the end there are conflicts and you have to update this pull request, check out the old version of this file. No need to add spaces here.

If there are no conflicts, then we'll just get spaces.

**Review by bbbales2 — DISMISSED** [review]
Good, merge ~~on the 20th~~, after the feature freeze

**Comment by rok-cesnovar** [comment]
Conflicts popped up when merging another generalization PR.

**Review by bbbales2 — APPROVED** [review]
Good!

**Pull request merged**